rudder-web/src/main/scala/com/normation/rudder/web/rest/RestQuicksearch.scala

```diff
@@ -52,7 +55,18 @@ class RestQuicksearch (
           toJsonError(None, e.messageChain)("quicksearch", false)
 
         case Full(results) =>
-          toJsonResponse(None, prepare(results, MAX_RES_BY_KIND))("quicksearch", false)
+          val filteredResults = {
+            import QuickSearchResultId._
+            results.filter { _.id match {
+              case QRNodeId(_)      => CurrentUser.checkRights(Read("node"))
+              case QRGroupId(_)     => CurrentUser.checkRights(Read("group"))
```

Comment:
you could test with: _:QRNodeID etc. "isInstanceOf" is amazingly fast on the jvm. 
On the same idea, it would be quicker to have val nodeOK = CurrentUser.checkRights(Read("node")) etc; and to put everything in a dedicated method "filterResultByRights(results, user)